cli/cdsctl/template_bulk.go

```diff
@@ -42,203 +60,475 @@ var templateBulkCmd = cli.Command{
+
+			// instance path should be formatted like MYPROJ/myWorkflow:myParameterKey=myValue
+			param := strings.Split(rawParams[i], "=")
+			if len(param) != 2 {
+				return nil, err
+			}
+			paramKey := strings.Split(param[0], ":")
+			if len(paramKey) != 2 {
+				return nil, err
+			}
+			instancePath := strings.Split(paramKey[0], "/")
+			if len(paramKey) != 2 {
+				return nil, err
+			}
+
+			params = append(params, templateBulkParameter{
+				InstancePath: templateBulkInstancePath{
+					ProjectKey:   instancePath[0],
+					WorkflowName: instancePath[1],
+				},
+				Key:   paramKey[1],
+				Value: param[1],
+			})
+		}
+	}
+
+	return params, nil
+}
+
+func templateExtractAndValidateFileParams(filePath string) (*sdk.WorkflowTemplate, []templateBulkParameter, error) {
+	if filePath == "" {
+		return nil, nil, nil
+	}
+
+	contentFile, format, err := exportentities.OpenPath(filePath)
+	if err != nil {
+		return nil, nil, err
+	}
+	defer contentFile.Close() //nolint
+
+	buf := new(bytes.Buffer)
+	if _, err := buf.ReadFrom(contentFile); err != nil {
+		return nil, nil, fmt.Errorf("cannot read from given file")
+	}
+
+	var f templateBulkFile
+	var errU error
+	switch format {
+	case exportentities.FormatJSON:
+		errU = json.Unmarshal(buf.Bytes(), &f)
+	case exportentities.FormatYAML:
+		errU = yaml.Unmarshal(buf.Bytes(), &f)
+	default:
+		return nil, nil, exportentities.ErrUnsupportedFormat
+	}
+	if errU != nil {
+		return nil, nil, fmt.Errorf("cannot unmarshal given file %v", err)
+	}
+
+	groupName, templateSlug, err := templateParsePath(f.TemplatePath)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	// try to get the template from cds
+	template, err := client.TemplateGet(groupName, templateSlug)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	params := []templateBulkParameter{}
+	for _, i := range f.Instances {
+		// instance path should be formatted like MYPROJ/myWorkflow
+		instancePath := strings.Split(i.WorkflowPath, "/")
+		if len(instancePath) != 2 {
+			return nil, nil, fmt.Errorf("invalid given instance path %s", i.WorkflowPath)
+		}
+
+		for _, value := range i.Parameters {
+			// instance path should be formatted like myParameterKey=myValue
+			param := strings.Split(value, "=")
+			if len(param) != 2 {
+				return nil, nil, err
+			}
+			params = append(params, templateBulkParameter{
+				InstancePath: templateBulkInstancePath{
+					ProjectKey:   instancePath[0],
+					WorkflowName: instancePath[1],
+				},
+				Key:   param[0],
+				Value: param[1],
+			})
+		}
+	}
+
+	return template, params, nil
+}
+
+func templateInitOperationFromParams(mwtis map[string]sdk.WorkflowTemplateInstance, minstances map[string]templateBulkInstancePath,
+	params, fileParams []templateBulkParameter) map[string]sdk.WorkflowTemplateBulkOperation {
+	// for all given instances, create an operation and reuse request if instance already exists
+	moperations := map[string]sdk.WorkflowTemplateBulkOperation{}
```

Comment:
indicate length of minstances

Reply:
I can't because moperations will be populated by minstances (instance given from params) but also from instances given in file (fileOperations) and params. I don't know what's the final size of the map.